langlib/lang.query/src/main/ballerina/helpers.bal

```diff
@@ -83,7 +84,16 @@ function getStreamFromPipeline(_StreamPipeline pipeline) returns stream<Type, Co
     return pipeline.getStream();
 }
 
-function toArray(stream<Type, CompletionType> strm, Type[] arr) returns Type[]|error {
+function toArray(stream<Type, CompletionType> strm, Type[] arr, boolean isReadOnly) returns Type[]|error {
+    if isReadOnly {
+        Type[] arr2 = [];
```

Comment:
can we add a commnet here why we create a temp arr. also may be rename it as `tempArr`